tests/sentry/api/endpoints/test_release_threshold_status.py

```diff
@@ -0,0 +1,391 @@
+        {
+            [release1]: {
+                [p1.id]: [threshold-p1-canary, threshold2-p1-canary, threshold-p1-prod]
+                [p2.id]: [threshold-p2-canary]
+            }
+        }
+        """
+        now = str(datetime.now())
+        yesterday = str(datetime.now() - timedelta(hours=24))
+        response = self.get_success_response(
+            self.organization.slug, start=yesterday, end=now, release_id=[self.release1.id]
+        )
+
+        assert len(response.data) == 1  # only fetched release1
+        assert len(response.data.get(self.release1.id)) == 2  # 2 projects (p1 & p2) in release 1
+
+        assert (
+            len(response.data.get(self.release1.id, {}).get(self.project1.id)) == 3
+        )  # p1 2x canary, 1x prod
+        assert (
+            response.data.get(self.release1.id, {})
+            .get(self.project1.id)[0]  # first threshold
+            .get("environment", {})
+            .get("name")
+            == self.canary_environment.name
+        )  # assert environment is 'canary'
+        assert (
+            response.data.get(self.release1.id, {})
+            .get(self.project1.id)[2]
+            .get("environment", {})
+            .get("name")
+            == self.production_environment.name
+        )  # assert environment is 'production'
+        assert (
+            len(response.data.get(self.release1.id, {}).get(self.project2.id)) == 1
+        )  # p2 1x canary
+        assert (
+            response.data.get(self.release1.id, {})
+            .get(self.project2.id)[0]  # first threshold
+            .get("environment", {})
+            .get("name")
+            == self.canary_environment.name
+        )  # assert environment is 'canary'
+
+        assert (
+            len(response.data.get(self.release2.id, {})) == 0
+        )  # release2 should not exist in the response
+
+    def test_get_success_project_id_filter(self):
+        """
+        Tests fetching thresholds within the past 24hrs filtered on environment
+
+        Set up creates
+        - 2 releases
+            - release1 - canary # env only matters for how we filter releases
+                - r1-proj1-canary # NOTE: is it possible to have a ReleaseProjectEnvironment without a corresponding ReleaseEnvironment??
+                - r1-proj2-canary
+            - release2 - prod # env only matters for how we filter releases
+                - r2-proj1-prod
+        - 4 thresholds
+            - project1 canary
+            - project1 canary
+            - project1 prod
+            - project2 canary
+
+        We'll filter for _only_ project2, so the response should look like
+        since project2 was only ever added to release1
+        {
+            [release1]: { # NOTE: fetches only the releases that include p2
+                [p1.id]: [threshold-p1-canary, threshold2-p1-canary, threshold-p1-prod]
+                [p2.id]: [threshold-p2-canary]
```

Comment:
when would we also want to know about p1 when we explicitly query for p2?

Reply:
Adding this note
> NOTE: in order to determine *release* health, we still need all projects (& thresholds) associated with that release
        So - filtering on project will give us all the releases associated with that project
        but we still need all the other projects associated with the release to determine health status